### DISCUSSION TITLE
Processing On Hold RAs

### DISCUSSION THREAD

**Discussion by uoced** [AdhocAdam/smletsexchangeconnector#314] [Q&A] [answered]
Greetings,

I have a requirement to process votes while review activities are on hold, any suggestions on how to use this solution to do so?

Thanks in advance!

**Comment by AdhocAdam** [comment, marked as answer]
Well I suppose the good news it that nothing stops you from processing an RA with the SMLets Exchange Connector _regardless_ of the Status it is in. So there isn't any reason you couldn't do this right now without any additional configuration of the _connector_!

However the challenge I suspect you may run/are running into (that is outside of the scope of the project), is that the Reviewers would need to know about that RAs existence, so they could vote, and the connector could in turn process it. The way this situation usually goes is that people have built custom Notifications when an RA goes from Pending -> In Progress. Reviewers are notified at the right time, and then it's on them to vote.

Since you want to process a vote when an RA is On Hold, you would need to conditionally trigger a notification when it goes On Hold so the Reviewers get the notification and thus the RA number, so they could in turn reply with [approved] and the SMLets Exchange Connector process that vote.

**Reply by uoced** [reply]
Thanks @AdhocAdam that's great news!  Voters currently receive an email once the RA is In Progress, which is placed On Hold when voting does not happen in a timely fashion.  The email is deleted by the default Exchange Connector but the vote is not processed while On Hold. Are you saying that the SMLets Exchange Connector will capture the vote regardless of status?  What happens after the votes are processed?  I need to be able to move On Hold RAs to Completed or Failed and parent work items to In Progress.

**Reply by AdhocAdam** [reply]
> ...which is placed On Hold when voting does not happen in a timely fashion.

Oh. That's interesting!

> The email is deleted by the default Exchange Connector but the vote is not processed while On Hold.'

I can't reproduce that. I just voted on an On Hold RA in my lab without issue. Make sure you have the Active Directory PowerShell module installed wherever you are running the connector as that is required to work with Review Activities.

> What happens after the votes are processed? I need to be able to move On Hold RAs to Completed or Failed and parent work items to In Progress.

And _that_ is the source of the issue you're describing. There are no out of box SCSM Workflows that say "Process RA's that are On Hold." The SMLets Exchange Connector simply processes the email/vote and then just moves onto the next email. It would be up to SCSM (well you in this case 😉) to build a custom Workflow that understands how to handle this scenario to shift an Activity/Parent Work Item accordingly.

**Reply by uoced** [reply]
> I can't reproduce that. I just voted on an On Hold RA in my lab without issue. Make sure you have the Active Directory PowerShell module installed wherever you are running the connector as that is required to work with Review Activities.

Are you testing with the SMLets Exchange Connector? I have not tried it yet, I was describing the behavior of the default Microsoft SCSM Exchange Connector.  Sorry for the confusion!

**Reply by AdhocAdam** [reply]
Yes, the SMLets Exchange Connector. The out of box connector does not support this functionality as you've observed.

**Reply by uoced** [reply]
Fantastic, I discovered this a few weeks ago and have been looking for a reason to make the switch.  Seems like I've found my reason!
